t/complex/si.xml

```diff
@@ -162,57 +208,63 @@
             <XMath>
-              <XMApp>
-                <XMTok meaning="times" role="MULOP">×</XMTok>
-                <XMTok meaning="0.3" role="NUMBER">0.3</XMTok>
-                <XMApp>
-                  <XMTok role="SUPERSCRIPTOP" scriptpos="post1"/>
+              <XMApp meaning="0.3E45">
+                <XMText meaning="times" role="MULOP" xml:id="S1.SS1.p1.m3.2">×</XMText>
+                <XMTok meaning="0.3" role="NUMBER" xml:id="S1.SS1.p1.m3.1">0.3</XMTok>
+                <XMApp xml:id="S1.SS1.p1.m3.3">
+                  <XMTok meaning="power" role="SUPERSCRIPTOP" scriptpos="post1"/>
                   <XMTok meaning="10" role="NUMBER">10</XMTok>
                   <XMTok fontsize="70%" meaning="45" role="NUMBER">45</XMTok>
                 </XMApp>
               </XMApp>
             </XMath>
-          </Math><break/><Math mode="inline" tex="1.654\times 2.34\times 3.430" text="1.654 * 2.34 * 3.430" xml:id="S1.SS1.p1.m4">
+          </Math><break/><Math mode="inline" tex="1.654\text{\times}2.34\text{\times}3.430" text="1.654 * 2.34 * 3.430" xml:id="S1.SS1.p1.m4">
             <XMath>
               <XMApp>
-                <XMTok meaning="times" role="MULOP">×</XMTok>
-                <XMTok meaning="1.654" role="NUMBER">1.654</XMTok>
-                <XMTok meaning="2.34" role="NUMBER">2.34</XMTok>
-                <XMTok meaning="3.430" role="NUMBER">3.430</XMTok>
+                <XMText meaning="times" role="MULOP" xml:id="S1.SS1.p1.m4.2">×</XMText>
+                <XMApp xml:id="S1.SS1.p1.m4.1">
+                  <XMText meaning="times" role="MULOP" xml:id="S1.SS1.p1.m4.1.2">×</XMText>
+                  <XMTok meaning="1.654" role="NUMBER" xml:id="S1.SS1.p1.m4.1.1">1.654</XMTok>
+                  <XMTok meaning="2.34" role="NUMBER" xml:id="S1.SS1.p1.m4.1.3">2.34</XMTok>
+                </XMApp>
+                <XMTok meaning="3.430" role="NUMBER" xml:id="S1.SS1.p1.m4.3">3.430</XMTok>
               </XMApp>
             </XMath>
-          </Math><break/><Math mode="inline" tex="\pi" text="pi" xml:id="S1.SS1.p1.m5">
+          </Math><break/><Math mode="inline" tex="\pi" text="\pi" xml:id="S1.SS1.p1.m5">
             <XMath>
-              <XMTok font="italic" name="pi" role="UNKNOWN">π</XMTok>
+              <XMTok font="italic" meaning="\pi" name="pi" role="UNKNOWN">π</XMTok>
             </XMath>
-          </Math><break/><Math mode="inline" tex="2\pi" text="2 * pi" xml:id="S1.SS1.p1.m6">
+          </Math><break/><Math mode="inline" tex="2\pi" text="2\pi" xml:id="S1.SS1.p1.m6">
             <XMath>
-              <XMApp>
-                <XMTok meaning="times" role="MULOP">⁢</XMTok>
-                <XMTok meaning="2" role="NUMBER">2</XMTok>
-                <XMTok font="italic" name="pi" role="UNKNOWN">π</XMTok>
-              </XMApp>
+              <XMDual>
+                <XMTok meaning="2\pi" role="NUMBER"/>
```

Comment:
A latex macro inside a meaning attribute? That... feels off somehow?

Reply:
I'm tempted to say `2\pi` is always an expression, rather than a number, unless it is some sort of strange embellished symbolic constant (which I don't think is the case here).

Reply:
Yeah, it's perverse. By default, siunitx treats both pi and ldots *essentially* as digits (and you can add more fake digits!).  Probably the right solution would be to detect this as (yet another) special case.